contracts/src/core/BatcherPaymentService.sol

```diff
@@ -28,6 +31,10 @@ contract BatcherPaymentService is
 
     mapping(address => uint256) public UserBalances;
 
+    // map to check if user has unlocked their funds
+    // 0 means funds are locked
+    mapping(address => uint256) public UserBalancesUnlockBlock;
+
     // map to check signature is only submitted once
     mapping(address => uint256) public UserNonces;
```

Comment:
At this point we should try mapping these 3 elements to 1 map `address -> struct{uint256, uint256, uint256}` containing this same information. 
I'm not 100% sure it will end up being cheaper, but I think it's worth a shot, it may be quite cheaper if implemented efficiently.

Reply:
Changed